### DISCUSSION TITLE
Using suspend in interface retrofit2 error : No JsonAdapter for retrofit2.Call

### DISCUSSION THREAD

**Discussion by reizamv** [square/retrofit#3941] [Q&A]
error happen when using suspend

 ```
@POST("permohonanUjiLab")
  suspend fun postpermohonanuji(@Body task: Reqpengajuanlab?): Call<responsepengajuanlab>
```
but it working if suspend remove

 ```
@POST("permohonanUjiLab")
  fun postpermohonanuji(@Body task: Reqpengajuanlab?): Call<responsepengajuanlab>
```

**Comment by JakeWharton** [comment]
For `suspend fun` usage, you can drop `Call` from the return type and only specify the body type.

So in your example:
```kotlin
@POST("permohonanUjiLab")
suspend fun postpermohonanuji(@Body task: Reqpengajuanlab?): responsepengajuanlab
```

The use of `suspend` is the execution mechanism and replaces `Call`, in the same way using something like RxJava's `Single` would replace `Call`.
